### PR TITLE
 SI-6502 Repl reset/replay take settings args

### PR DESCRIPTION
**Pull request by som-snytt**
The reset and replay commands take arbitrary command line args.
When settings args are supplied, the compiler is recreated.

For uniformity, the settings command performs only the usual
arg parsing: use `-flag:true` or `-flag` instead of `+flag`, and clearing a
setting is promoted to the command line, so that -Xlint: is not
an error but clears the flags.

```
scala> maqicode.Test main null
<console>:8: error: not found: value maqicode
              maqicode.Test main null
              ^

scala> :reset -classpath/a target/scala-2.11/sample_2.11-1.0.jar
Resetting interpreter state.
Forgetting all expression results and named terms: $intp

scala> maqicode.Test main null
Hello, world.

scala> val i = 42
i: Int = 42

scala> s"$i is the loneliest numbah."
res1: String = 42 is the loneliest numbah.

scala> :replay -classpath ""
Replaying: maqicode.Test main null
Hello, world.

Replaying: val i = 42
i: Int = 42

Replaying: s"$i is the loneliest numbah."
res1: String = 42 is the loneliest numbah.

scala> :replay -classpath/a ""
Replaying: maqicode.Test main null
<console>:8: error: not found: value maqicode
              maqicode.Test main null
              ^

Replaying: val i = 42
i: Int = 42

Replaying: s"$i is the loneliest numbah."
res1: String = 42 is the loneliest numbah.
```

Clearing a clearable setting:

```
scala> :reset -Xlint:missing-interpolator
Resetting interpreter state.

scala> { val i = 42 ; "$i is the loneliest numbah." }
<console>:8: warning: possible missing interpolator: detected interpolated identifier `$i`
              { val i = 42 ; "$i is the loneliest numbah." }
                             ^
res0: String = $i is the loneliest numbah.

scala> :reset -Xlint:
Resetting interpreter state.
Forgetting this session history:

{ val i = 42 ; "$i is the loneliest numbah." }

scala> { val i = 42 ; "$i is the loneliest numbah." }
res0: String = $i is the loneliest numbah.

scala> :replay -Xlint:missing-interpolator
Replaying: { val i = 42 ; "$i is the loneliest numbah." }
<console>:8: warning: possible missing interpolator: detected interpolated identifier `$i`
              { val i = 42 ; "$i is the loneliest numbah." }
                             ^
res0: String = $i is the loneliest numbah.
```

Review by @gkossakowski

### CODE REVIEW

**Comment by som-snytt** [comment]
PLS REBUILD/pr-scala@13e9317

File: src/repl/scala/tools/nsc/interpreter/ILoop.scala [hunk 1]
```diff
@@ -220,8 +220,9 @@ class ILoop(in0: Option[BufferedReader], protected val out: JPrintWriter)
     cmd("paste", "[-raw] [path]", "enter paste mode or paste a file", pasteCommand),
     nullary("power", "enable power user mode", powerCmd),
     nullary("quit", "exit the interpreter", () => Result(keepRunning = false, None)),
-    nullary("replay", "reset execution and replay all previous commands", replay),
-    nullary("reset", "reset the repl to its initial state, forgetting all session entries", resetCommand),
+    cmd("replay", "[-cp <path>]", "reset the repl and replay all previous commands", replayCommand),
+    //cmd("require", "<path>", "add a jar or directory to the classpath", require),  // TODO
+    cmd("reset", "[-cp <path>]", "reset the repl to its initial state, forgetting all session entries", resetCommand),
```
Comment (som-snytt) [inline src/repl/scala/tools/nsc/interpreter/ILoop.scala]: doesn't reflect arbitrary settings.

**Comment by som-snytt** [comment]
Rebased and corrected `:help`.

**Comment by som-snytt** [comment]
PLS REBUILD/pr-scala@e720bab

**Comment by adriaanm** [comment]
ping @gkossakowski

**Comment by gkossakowski** [comment]
LGTM

**Comment by gkossakowski** [comment]
Excellent work!

**Pull request merged**